docs/develop/plugins-guide.mdx

```diff
@@ -0,0 +1,351 @@
+---
+id: plugins-guide
+title: Plugins Guide
+sidebar_label: Plugins guide
+description: Best practices for creating plugins for AI Agents
+toc_max_heading_level: 4
+hide_table_of_contents: true
+keywords:
+ - ai agents
+ - best practices
+tags:
+ - Best Practices
+ - AI Agents
+---
+
+import SdkTabs from '@site/src/components';
+
+# Plugins
+
+A **Plugin** is an abstraction that will let your users add your feature to their Workflows in one line. It makes it seamless for platform developers to add their own custom functionality to many Workflows. Using plugins, you can build reusable open source libraries or build add-ons for engineers at your company.
+
+Here are some common use cases for plugins:
+
+- AI Agent SDKs
+- Observability, tracing, or logging middleware
+- Adding reliable built-in functionality such as LLM calls, corporate messaging, and payments infrastructure
+- Encryption or compliance middleware
+
+## How to build a Plugin
+
+The recommended way to start building plugins is with a `SimplePlugin`. This abstraction will tackle the vast majority of plugins people want to write.  
+
+For advanced use cases, you can extend the methods in lower-level classes that Simple Plugin is based on without re-implementing what you’ve done. See the [Advanced Topics section](#advanced-topics-for-plugins) for more information.
+
+### Example Plugins
+
+If you prefer to learn by getting hands-on with code, check out some existing plugins.
+
+- Temporal's Python SDK ships with an [OpenAI Agents SDK](https://github.com/temporalio/sdk-python/tree/main/temporalio/contrib/openai_agents) plugin
+- Pydantic plugin when it’s fixed
+
+## What you can provide to users in a plugin
+
+There are a number of features you can give your users with a plugin. Here's a short list of some of the things you can do and more are being added.
+
+- [Built-in Activities](#built-in-activity)
+- [Workflow libraries](#workflow-libraries)
+- [Built-in Child Workflows](#built-in-child-workflows)
+- [Built-in Nexus Operations](#built-in-nexus-operations)
+- [Custom Data Converters](#custom-data-converters)
+- [Interceptors](#interceptors)
+
+### Built-in Activity
+
+You can provide built-in Activities in a Plugin for users to call from their Workflows. Activities are the most common Temporal primitive and should be scoped to:
+
+- Single write operations
+- Batches of similar writes
+- One or more read operations followed by a write operation
+- A read that should be memoized, like an LLM call, a large download, or a slow-polling read
```

Comment:
This reads awkwardly to me. Each line sort of contradicts the last. Presumably we have advice elsewhere in the docs we can link to about what an activity should be scoped to?

This could be  just "Activities should be scoped to a small, ideally idempotent, unit of work" or something